### PR TITLE
feat(connector): [Elavon] Add card payments, psync, refunds, rsync and void flows

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Elavon Inc., formerly NOVA, is a processor of card transactions and a subsidiary of U.S. Bancorp. Elavon offers merchant processing in more than 30 countries and supports the payment needs of more than 1,000,000 merchant locations across the globe.

Resolves [#2957](https://github.com/juspay/hyperswitch-cloud/issues/2957)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Can't be tested as Elavon has a closed sandbox.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/elavon.rs [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+        &self,
+        req: &types::PaymentsCaptureRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok(format!(
+            "{}/{}/{}",
+            self.base_url(connectors),
+            "transactions",
+            req.request.connector_transaction_id
+        ))
```
Comment (SamraatBansal) [inline crates/router/src/connector/elavon.rs:346]:
```suggestion
        Ok(format!(
            "{}/transactions/{}",
            self.base_url(connectors),
            req.request.connector_transaction_id
        ))
```

File: crates/router/src/connector/elavon/transformers.rs [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+            }),
+            None => Err(errors::ConnectorError::MissingRequiredField {
+                field_name: "address",
+            })?,
+        }
+    }
+}
+
+pub struct ElavonAuthType {
+    pub(super) api_key: Secret<String>,
```
Comment (SamraatBansal) [inline crates/router/src/connector/elavon/transformers.rs:177]: if its a `BodyKey` 2 key fields should be present in this struct

File: crates/router/src/connector/elavon/transformers.rs [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    Authorized,
+    Captured,
+    Voided,
+    Settled,
+    Expired,
+    SettlementDelayed,
+    Rejected,
+    HeldForReview,
+    #[default]
+    Unknown,
```
Comment (SamraatBansal) [inline crates/router/src/connector/elavon/transformers.rs:209]: Why have we set default value to unknown?

File: crates/router/src/connector/elavon/transformers.rs [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+            ElavonStatus::Captured => Self::Pending,
+            ElavonStatus::Voided => Self::Failure,
+            ElavonStatus::Settled => Self::Success,
+            ElavonStatus::Expired => Self::Failure,
+            ElavonStatus::SettlementDelayed => Self::Pending,
+            ElavonStatus::Rejected => Self::Failure,
+            ElavonStatus::HeldForReview => Self::ManualReview,
+            ElavonStatus::Unknown => Self::Pending, // Have to get confirmation from the connector
+            ElavonStatus::AuthorizationPending => Self::Pending,
+        }
```
Comment (SamraatBansal) [inline crates/router/src/connector/elavon/transformers.rs:304]:
```suggestion
            ElavonStatus::Declined | ElavonStatus::Voided |  ElavonStatus::Expired | ElavonStatus::Rejected => Self::Failure,
            ElavonStatus::Settled => Self::Success,
            ElavonStatus::HeldForReview => Self::ManualReview,
            ElavonStatus::Unknown => Self::Pending, // Have to get confirmation from the connector
            ElavonStatus::AuthorizationPending | ElavonStatus::SettlementDelayed | ElavonStatus::Captured | ElavonStatus::Authorized  => Self::Pending,
        }
```

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Comment by AkshayaFoiger** [comment]
Elavon converge, no more supports REST API integration. For further information refer [here](https://developer.elavon.com/products/converge/v1/overview).

**Comment by JacobHayes** [comment]
> Blocking this Pull Request as [Elavon converge](https://developer.elavon.com/products/converge/v1/overview), no more supports REST API integration.

I'm not sure what the API was before, but it seems they have an XML (😢) API in case that'd be possible to use.

**Comment by AkshayaFoiger** [comment]
Connector contract changed

**Pull request closed without merging**